doc/sphinx/README.rst

```diff
@@ -64,7 +64,26 @@ The signatures of most objects can be written using a succinct DSL for Coq notat
   an alternative, indicating than one of multiple constructs can be used
 
 ``%{``, ``%}``, ``%|``
-  an escaped character (rendered without the leading ``%``).  In most cases, escaping is not necessary.  In particular, the following expressions are all parsed as plain text, and do not need escaping: ``{ xyz }``, ``x |- y``.  But the following escapes *are* needed: ``{| a b %| c | d }``, ``all: %{``.  (We use ``%`` instead of the usual ``\`` because you'd have to type ``\`` twice in your reStructuredText file.)
+  an escaped character (rendered without the leading ``%``).  We use ``%``
+  instead of the usual ``\`` because reStructuredText would require using
+  the awkward double escape ``\\``.
+
+  - Use it to escape grammar literal strings that are the same as metasyntax,
+  such as ``{``, ``|``, ``}``, ``{|`` and ``{+,``.  (While this is optional for
+  ``|`` and ``{ ... }`` outside of ``{| ... }``, always using the escape
+  requires less thought.)
+
+  - Literals such as ``|-`` and ``||`` don't need to be escaped.
+
+  - The literal ``%`` shouldn't be escaped.
+
+  - Don't use the escape for a ``|`` separator in ``{*`` and ``{+``.  These
+    should appear as ``{*|`` and ``{+|``.
+
+  - At this writing, the sequence ``{| {+, foo } }`` gives a parse error.
+
+  - At this writing, the sequence ``{+|| xx }`` gives a parse error, while
```

Comment:
I'm not a fan of documenting this we perceive as bugs; if we don't like them, let's fix them; telling people in the readme: this gives a parse error doesn't help much, I think.